django/db/backends/oracle/base.py

```diff
@@ -475,23 +468,17 @@ def _fix_for_params(self, query, params, unify_by_values=False):
             params_dict = {param: ':arg%d' % i for i, param in enumerate(set(params))}
             args = [params_dict[param] for param in params]
             params = dict(zip(params_dict.values(), params_dict.keys()))
-            query = convert_unicode(query % tuple(args), self.charset)
+            query = force_text(query % tuple(args), self.charset)
         else:
             # Handle params as sequence
             args = [(':arg%d' % i) for i in range(len(params))]
-            query = convert_unicode(query % tuple(args), self.charset)
+            query = force_text(query % tuple(args), self.charset)
         return query, self._format_params(params)
```

Comment:
@felixxm It looks like we can't rely on hashing on Python 2 as `hash(b'foo') == hash(u'foo')`.

May I suggest you use `(value, type(value))` as `params` key instead of simply `value`?